src/test/TimelockHook.t.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {DSTest} from "ds-test/src/test.sol";
+import {Hevm} from "solmate/test/utils/Hevm.sol";
+import {IAludelV3} from "../contracts/aludel/AludelV3.sol";
+
+import {TimelockHook} from "../contracts/aludel/TimelockHook.sol";
+
+contract PowerSwitchTest is DSTest {
+    Hevm private cheats;
+
+    function setUp() public {
+        cheats = Hevm(HEVM_ADDRESS);
+    }
+
+    function test_30_days() public {
+        uint64 timestamp = uint64(block.timestamp);
+        TimelockHook hook = new TimelockHook(30 days);
+        cheats.expectRevert(TimelockHook.TimelockNotElapsed.selector);
+        hook.unstakeAndClaimPost(
+            IAludelV3.StakeData({amount: 100, timestamp: timestamp})
+        );
+        cheats.warp(30 days+1);
+        hook.unstakeAndClaimPost(
+            IAludelV3.StakeData({amount: 100, timestamp: timestamp})
+        );
+    }
+
+    function test_100_days() public {
```

Comment:
You can add a uint64 param to the test's signature and it will try different values.

Check this out: https://book.getfoundry.sh/forge/fuzz-testing

Reply:
implemented :sparkles: